MathLib/Curve/PiecewiseLinearCurve.cpp

```diff
@@ -0,0 +1,87 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ * \file   PiecewiseLinearCurve.cpp
+ *
+ * Created on November 11, 2016, 10:49 AM
+ */
+
+#include "PiecewiseLinearCurve.h"
+
+#include <algorithm>
+#include <cmath>
+#include <limits>
+
+namespace MathLib
+{
+bool PiecewiseLinearCurve::isMonotonic() const
+{
+    const double gradient0 = getDerivative(_supp_pnts[0]);
+
+    if (std::fabs(gradient0) < std::numeric_limits<double>::min())
+        return false;
+    else
+    {
+        return std::none_of(_supp_pnts.begin(), _supp_pnts.end(),
+                            [&](const double p) {
+                                return this->getDerivative(p) * gradient0 <= 0.;
+                            });
+    }
+}
+
+double PiecewiseLinearCurve::getVariable(const double y) const
+{
+    std::size_t interval_idx = 0;
```

Comment:
Thank. I revised accordingly. However, I got another idea: create two classes as
PiecewiseLinearCurve
PiecewiseLinearMonotonicCurve

With that, the monotonicity can only assert in ctor. To do this, I have to wait until  PR #1561 is merged.

@TomFischer  How do you think about this idea?